src/components/icon/index.tsx

```diff
@@ -54,12 +65,48 @@ const Icon = forwardRef((props: Props, ref: any) => {
     assetName,
     modifiers,
     recorderTag,
+    badgeProps,
+    hideBadge,
     ...others
   } = props;
   const {margins} = modifiers;
+  const shouldShowBadge = !hideBadge && badgeProps;
   const iconSize = size ? {width: size, height: size} : undefined;
   const shouldFlipRTL = supportRTL && Constants.isRTL;
 
+  const getBadgeStyling = (): StyleProp<ViewStyle> => {
+    const containerStyle = badgeProps?.containerStyle;
+    const badgePosition: StyleProp<ViewStyle> = {
+      position: 'absolute',
+      top: -3
+    };
+    if (isPimple()) {
+      badgePosition.right = -2;
+      badgePosition.top = -2;
+    } else {
+      badgePosition.right = -5;
+      badgePosition.top = -5;
+    }
+    return [badgePosition, containerStyle];
+  };
+
+  const getBadgeLabel = () => {
+    const {badgeProps} = props;
+    const label = get(badgeProps, 'label');
+    return label;
+  };
+
+  const getBadgeSize = () => {
+    const size = isPimple() ? 10 : 16;
```

Comment:
Why are you setting a default size? The badge and the pimple already have a default, and the user can pass it in the badgeProps if needed